enclave/src/lib.rs

```diff
@@ -536,6 +539,69 @@ pub unsafe extern "C" fn run_db_thread() -> sgx_status_t {
     }
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn send_cid(
```

Comment:
I know this is not yet finished, but could you move send_cid to a separate file? lib.rs is big & ugly enough as it is, no need to further blow it up.